reframe/core/systems.py

```diff
@@ -13,6 +13,169 @@
 from reframe.core.environments import (Environment, ProgEnvironment)
 
 
+class Processor(jsonext.JSONSerializable):
+    '''A representation of a processor inside ReFrame.
+
+    .. warning::
+       Users may not create :class:`Processor` objects directly.
+    '''
+
+    def __init__(self, processor_info):
+        self._arch = None
+        self._num_cpus = None
+        self._num_cpus_per_core = None
+        self._num_cpus_per_socket = None
+        self._num_sockets = None
+        self._topology = None
+        self._info = processor_info
+
+        if processor_info == {}:
+            return
+
+        for var in ['arch', 'num_cpus', 'num_cpus_per_core',
+                    'num_cpus_per_socket', 'num_sockets', 'topology']:
```

Comment:
You don't need to reiterate on that. I'd rather restrict the JSON schema not to allow additional properties, so that you can set the attributes by simply iterating over the `processor_info` dictionary.